routing/index_graph.hpp

```diff
@@ -29,23 +32,129 @@ class JointEdge final
 
 private:
   // Target is vertex going to for outgoing edges, vertex going from for ingoing edges.
-  Joint::Id const m_target;
-  double const m_weight;
+  Joint::Id m_target;
+  double m_weight;
+};
+
+class JointEdgeGeom final
+{
+public:
+  JointEdgeGeom() = default;
+  JointEdgeGeom(Joint::Id target, vector<RoadPoint> const & path) : m_target(target), m_path(path)
+  {
+  }
+  Joint::Id GetTarget() const { return m_target; }
+  vector<RoadPoint> const & GetPath() const { return m_path; }
+
+private:
+  // Target is vertex going to for outgoing edges, vertex going from for ingoing edges.
+  Joint::Id m_target = Joint::kInvalidId;
+  vector<RoadPoint> m_path;
+};
+
+class DirectedEdge final
```

Comment:
I added the comment.

As for JointEdge:
- JointEdge is TEdgeType. That means an edge in an edge in A* algorithm.
- TEdgeType is use for building route. In that case it's not important knowing which feature is used because we're interested only in shortest path;
- Now it's field: Joint::Id m_target could represent ingoing or outgoing edge depending on the situation;
- To exchange DirectedEdge with JointEdge we need to memorize weight in DirectedEdge. So the size of DirectedEdge will be 20 bytes while size of JointEdge is only 12.

Generally speaking I guess it's possible to exchange JointEdge with JointEdge but it's not easy and I'm not sure we get a lot advantages from it.